src/test/components/TimelineMarkers.test.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React from 'react';
+import TimelineMarkers from '../../content/containers/TimelineMarkers';
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+import mockCanvasContext from '../fixtures/mocks/canvas-context';
+import { storeWithProfile } from '../fixtures/stores';
+import { getProfileWithMarkers } from '../store/fixtures/profiles';
+
+jest.useFakeTimers();
+
+it('renders TimelineMarkers correctly', () => {
+  window.requestAnimationFrame = fn => setTimeout(fn, 0);
+  window.devicePixelRatio = 1;
+  const ctx = mockCanvasContext();
+
+  /**
```

Comment:
This is definitely the most brittle part of the test. Not super happy about mocking things out like this, but it lets us use `ref`s.